lib/broccoli_sass_compiler.js

```diff
@@ -543,81 +530,67 @@ module.exports = class BroccoliSassCompiler extends BroccoliPlugin {
    * as an array of pairs (filename, hash).
    */
   hashDependencies(details) {
-    let dependencyPromises = [];
+    let depsWithHashes = [];
 
     this.dependenciesOf(details.fullSassFilename).forEach(f => {
-      dependencyPromises.push(this.hashForFile(f).then(h =>{
+      try {
+        let h = this.hashForFile(f);
+
         if (f.startsWith(details.srcPath)) {
           f = f.substring(details.srcPath.length + 1);
         }
-        return [
+        depsWithHashes.push([
           f,
           h.digest("hex")
-        ];
-      }).catch(e => {
+        ]);
+      } catch (e) {
         if (typeof e === "object" && e !== null && e.code === "ENOENT") {
           persistentCacheDebug("Ignoring non-existent file: %s", f);
-          return [];
+          depsWithHashes.push([]);
         } else {
           throw e;
         }
-      }));
+      }
     });
 
-    return RSVP.all(dependencyPromises).then(depsWithHashes => {
-      // prune out the dependencies that weren't files.
-      return depsWithHashes.filter(dwh => dwh.length > 0);
-    });
+    // prune out the dependencies that weren't files.
+    return depsWithHashes.filter(dwh => dwh.length > 0);
   }
 
   /* read all output files asynchronously and return the contents
    * as a hash of relative filenames to base64 encoded strings.
    */
   readOutputs(details) {
-    let readPromises = [];
+    let reads = [];
     let outputs = this.outputsFrom(details.fullSassFilename);
 
-    outputs.forEach(output =>  {
-      readPromises.push(readFile(output).then(buffer => {
-        return [output, buffer.toString("base64")];
-      }));
-    });
-
-    return RSVP.all(readPromises).then(outputs => {
-      let outputContents = {};
+    outputs.forEach(output => reads.push([output, fs.readFileSync(output, "base64")]));
```

Comment:
Probably easier as `let reads = outputs.map`?

Reply:
ya I would but `outputs` is a set, which has no `map`

Reply:
ah! I didn't follow that backwards to see what it was..